### DISCUSSION TITLE
Enhancing the --chat feature

### DISCUSSION THREAD

**Discussion by schielb** [TheR1D/shell_gpt#241] [Ideas]
There are two ideas that I think could make this tool a lot more user-friendly, specifically when handling previous chats that you have had with the bot:

- When you type `sgpt --chat`, have a feature where you can press `Tab` that will pull up all previous chats you have had - then you `Tab` through the list to pick the one you want.
- To avoid constantly having to type in `sgpt --chat <chat_name> "Message"`, it would be nice to have a feature where you just type in `sgpt --chat <chat_name>` and essentially a new shell opens up, like the python shell when you just type `python3` into the terminal. This way you can make the conversation continue without having to execute a new command for each query you create as part of that chat.
  - If you want to start a new chat from scratch, you can just use a new `<chat_name>` in the command.

**Comment by schielb** [comment]
I'd be down to try and help with this; where in the code does handling the `--chat` arg take place?

**Comment by schielb** [comment]
Hm, re-reading through the documentation, I realize that the second point is already figured out with REPL mode... my bad 😂

**Comment by initialvisuals** [comment]
I wrote some fish shell commands which make working with shell-gpt easier.

alias ai="sgpt"

function hey
    set -l query (string join " " $argv)
    ai --chat MyChat "$query"
end

function cmd
    set -l query (string join " " $argv)
    ai --shell "$query"
end


these now let you use your specified chat automatically or quickly enter shell commands like this with no quotes:

hey give me a Python script to do that

cmd open an Apache server port 8080

hey do you remember how to find running daemons
